src/lib/y2packager/release_notes_fetchers/url.rb

```diff
@@ -0,0 +1,360 @@
+        pos = relnotes_url.rindex("/")
+        @relnotes_url_base = relnotes_url[0, pos]
+      end
+
+      # Return release notes filename including language and format
+      #
+      # @return [String] Release notes filename
+      def release_notes_file(user_lang, format)
+        "RELEASE-NOTES.#{user_lang}.#{format}"
+      end
+
+      # curl proxy options
+      #
+      # @return [String] to be interpolated in a .target.bash command, unquoted
+      def curl_proxy_args
+        return @curl_proxy_args if @curl_proxy_args
+        @curl_proxy_args = ""
+        # proxy should be set by inst_install_inf if set via Linuxrc
+        Yast::Proxy.Read
+        # Test if proxy works
+
+        return @curl_proxy_args unless Yast::Proxy.enabled
+        # it is enough to test http proxy, release notes are downloaded via http
+        proxy_ret = Yast::Proxy.RunTestProxy(
+          Yast::Proxy.http,
+          "",
+          "",
+          Yast::Proxy.user,
+          Yast::Proxy.pass
+        )
+
+        http_ret = proxy_ret.fetch("HTTP", {})
+        if http_ret.fetch("tested", true) == true && http_ret.fetch("exit", 1) == 0
+          user_pass = Yast::Proxy.user != "" ? "#{Yast::Proxy.user}:#{Yast::Proxy.pass}" : ""
+          proxy = "--proxy #{Yast::Proxy.http}"
+          proxy << " --proxy-user '#{user_pass}'" unless user_pass.empty?
+        end
+
+        @curl_proxy_args = proxy
+      end
+
+      # Release notes index for the given product
+      #
+      # @return [Array<String>] Index containing the list of release notes files
+      # @see #download_release_notes_index
+      def release_notes_index
+        return @release_notes_index if @release_notes_index
+        @release_notes_index = download_release_notes_index(relnotes_url_base) || []
+      end
+
+      # Determine whether the release notes index contains an entry for the given
+      # language and format
+      #
+      # @return [Boolean]
+      def indexed_release_notes_for?(user_lang, format)
+        release_notes_index.include?(release_notes_file(user_lang, format))
+      end
+
+      # Download of index of release notes for a specific product
+      #
+      # @param url_base URL pointing to directory with the index
+      # @return [Array<String>,nil] filenames, nil if not found
+      def download_release_notes_index(url_base)
+        url_index = url_base + "/directory.yast"
+        log.info("Index with available files: #{url_index}")
+        tmpfile = Tempfile.new("directory.yast-")
+
+        # download the index with much shorter time-out
+        ok = curl_download(url_index, tmpfile.path, max_time: 30)
+
+        if ok
+          log.info("Release notes index downloaded successfully")
+          index_file = File.read(tmpfile.path)
+          if index_file.nil? || index_file.empty?
+            log.info("Release notes index empty, not filtering further downloads")
+            nil
+          else
+            rn_filter = index_file.split("\n")
```

Comment:
NP: `rn_filter` sounds like a Proc for filtering release notes, I'd use `rn_index` or something like that here.